pimd/pim_nht.c

```diff
@@ -483,6 +483,37 @@ static int pim_update_upstream_nh(struct pim_instance *pim,
 	return 0;
 }
 
+static int pim_upstream_nh_if_update_helper(struct hash_bucket *bucket,
+					    void *arg)
+{
+	struct pim_nexthop_cache *pnc = bucket->data;
+	struct pnc_hash_walk_data *pwd = arg;
+	struct pim_instance *pim = pwd->pim;
+	struct interface *ifp = pwd->ifp;
+	struct nexthop *nh_node = NULL;
+	ifindex_t first_ifindex;
+
+	for (nh_node = pnc->nexthop; nh_node; nh_node = nh_node->next) {
+		first_ifindex = nh_node->ifindex;
+		if (ifp != if_lookup_by_index(first_ifindex, pim->vrf->vrf_id))
```

Comment:
The logic here is… super confusing/odd. Either the ifindex is the same for all `pnc->nexthop` items, then the check could be outside the loop. Or the ifindex isn't the same, then the check is wrong…

Reply:
Thanks for the review.
I will update the PR.

Reply:
@eqvinox Thanks for your comment.
if-index is not same for pnc->nexthop, this is reason check inside the loop.